### PR TITLE
Initial CI

### PR DESCRIPTION
**Pull request by zmoon**
* [x] pre-commit config with `findent` and whitespace stuff
* [x] apply whitespace stuff to non-new files, but not the indent
* [x] initial CMake config to compile the new files
* [x] GitHub Actions workflows to run pre-commit and do the build with different GFortran versions 
* [x] document how to set up pre-commit and do the build locally
* [x] add similar debug flags for Intel (Intel on GitHub Actions is easy for now since we don't need any deps yet)
* [x] a few small name changes to get things to compile (+ commented out atomic number species thing)

### CODE REVIEW

File: src/core/yaml_parser.F90
Comment (zmoon) [inline src/core/yaml_parser.F90:1]: @bbakernoaa we probably want to change the file name to match the module name

Comment (bbakernoaa) [reply src/core/yaml_parser.F90:1]: Let’s do it

Comment (bbakernoaa) [reply src/core/yaml_parser.F90:1]: Do you wanna go ahead and merge this?

**Comment by bbakernoaa** [comment]
@zmoon I'm going to merge this unless you say otherwise

File: src/core/yaml_parser.F90
Comment (zmoon) [reply src/core/yaml_parser.F90:1]: Can you change this file name? And in the corresponding cmakelists file too?

Comment (bbakernoaa) [reply src/core/yaml_parser.F90:1]: should be good now

Comment (zmoon) [reply src/core/yaml_parser.F90:1]: Can you rename the exclude in the pre commit config as well? Sorry. Otherwise lint fails.

Comment (zmoon) [reply src/core/yaml_parser.F90:1]: Resolved in 6a4e50f

**Comment by zmoon** [comment]
@bbakernoaa ready to merge if you approve

**Pull request merged**